File: 区间/56.合并区间.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) <= 1:
            return intervals

        # 对区间进行排序
        intervals.sort(key=lambda x: x[0])
        merged = [intervals[0]]
        for interval in intervals[1:]:
            if interval[0] <= merged[-1][1]:
                # 如果当前区间与前一个合并后的区间重叠，进行合并
                merged[-1][1] = max(merged[-1][1], interval[1])
            else:
                # 如果没有重叠，将当前区间添加到合并后的区间列表
                merged.append(interval)
        return merged

File: 区间/56.合并区间.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    if (intervals.length < 2) return intervals;
    // 按照每个元素的第一项排序
    intervals.sort((a, b) => a[0] - b[0]);  // a - b递增
    //* 在原数组上合并比较复杂，可以创建新数组
    const merged = [intervals[0]];
    for (let i = 1; i < intervals.length; i++) {
        const mergedLastIndex = merged.length - 1;
        //! 处理 合并/不合并 两种情况
        if (intervals[i][0] <= merged[mergedLastIndex][1]) {
            merged[mergedLastIndex][1] = Math.max(intervals[i][1], merged[mergedLastIndex][1]);
        } else {
            merged.push(intervals[i]);
        }
    }
    return merged;
};